plugin/proxy/dns.go

```diff
@@ -93,7 +93,9 @@ func exchange(m *dns.Msg, co net.Conn) (*dns.Msg, error) {
 
 	writeDeadline := time.Now().Add(defaultTimeout)
 	dnsco.SetWriteDeadline(writeDeadline)
-	dnsco.WriteMsg(m)
+	if err := dnsco.WriteMsg(m); err != nil {
```

Comment:
this looks like an unrelated change

Reply:
Yes, I realized the error was not propagated up the stack when I was testing this thing. Would you like me to send a separate PR for this?

Reply:
yep, please do. Should probably also be `log.Debug`